build.gradle.kts

```diff
@@ -111,23 +111,94 @@ subprojects {
     apply(plugin = "publishing-conventions")
 }
 
+// == publishing setup ==
+val testRepositoryDependency = configurations.create("testRepository") {
+    isVisible = true
+    isCanBeResolved = false
+    isCanBeConsumed = false
+}
+
+val mergeProject = project
+
+subprojects {
+    if (name in unpublishedProjects) return@subprojects
+    apply(plugin = "publishing-conventions")
+    mergeProject.dependencies.add(testRepositoryDependency.name, this)
+}
+
+
+
+val testRepositories = configurations.create("testRepositories") {
+    isVisible = false
+    isCanBeResolved = true
+    // this config consumes modules from OTHER projects, and cannot be consumed by other projects
+    isCanBeConsumed = false
+
+    attributes {
+        attribute(Attribute.of("kotlinx.serialization.repository", String::class.java), "test")
+    }
+    extendsFrom(testRepositoryDependency)
+}
+
+tasks.register<ArtifactsCheckTask>("checkArtifacts") {
+    repositories.from(testRepositories)
+}
+
+abstract class ArtifactsCheckTask: DefaultTask() {
```

Comment:
I think this should be in a separate file rather than in build.gradle.kts

Reply:
Also, perhaps we can move `artifacts.txt` from root folder to `gradle` or something